### PR TITLE
fix(sql): fix COPY of file with duplicate column names

### PR DESCRIPTION
**Pull request by bziobrowski**
Fixes copy breaking on duplicate column names in detected or enforced file header line . 
Also - tries to avoid clash between existing and synthetic column names .

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|1|1|100.00%|

File: core/src/main/java/io/questdb/cutlass/text/TextMetadataDetector.java [hunk 1]
```diff
@@ -93,12 +95,26 @@ public void evaluateResults(long lineCount, long errorCount) {
...
+                for (int attempt = 0; attempt < 20; attempt++) {
+                    if (!uniqueColumnNames.contains(colName)) {
+                        break;
+                    }
+
+                    tempSink.put('_');
+                    colName = tempSink.toString();
+                }
+
+                columnNames.setQuick(i, colName);
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cutlass/text/TextMetadataDetector.java:113]: What happens if there are > 20 columns with the same name?

File: core/src/main/java/io/questdb/cutlass/text/TextMetadataDetector.java [hunk 2]
```diff
@@ -93,12 +95,26 @@ public void evaluateResults(long lineCount, long errorCount) {
...
-                columnNames.setQuick(i, tempSink.toString());
+                String colName = tempSink.toString();
+
+                for (int attempt = 0; attempt < 20; attempt++) {
+                    if (!uniqueColumnNames.contains(colName)) {
+                        break;
+                    }
+
+                    tempSink.put('_');
+                    colName = tempSink.toString();
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cutlass/text/TextMetadataDetector.java:110]: you can move this line out of the loop, and test `contains` using the sink

Comment (bziobrowski) [reply core/src/main/java/io/questdb/cutlass/text/TextMetadataDetector.java]: Fixed

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/std/ObjList.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|1|1|100.00%|

File: core/src/main/java/io/questdb/cutlass/text/TextMetadataDetector.java (hunk 1 above)
Comment (bziobrowski) [reply core/src/main/java/io/questdb/cutlass/text/TextMetadataDetector.java]: Now it will throw an exception if it fails to generate unique column name .

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/std/ObjList.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|1|1|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/std/ObjList.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/std/ObjList.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|1|1|100.00%|

**Pull request merged**